pandas/core/groupby/base.py

```diff
@@ -93,19 +94,21 @@ def _gotitem(self, key, ndim, subset=None):
                                    'idxmax'])
 
 
-def whitelist_method_generator(base, klass, whitelist):
+def whitelist_method_generator(base: 'Type[GroupBy]',
```

Comment:
While this is probably just used for GroupBy probably OK to just type this as `Type`, i.e. it could theoretically accept any Type not just a GroupBy

Reply:
this is pretty descriptive here, if the more specific types work that is good

Reply:
OK sure. Though I think should be `Type['GroupBy']` then (note difference in quoting)

Reply:
My preference is an all-or-nothing approach to quoting. Else it's more work to check if a annotation is quoted or not.

Reply:
Not sure I totally follow your point but the intention of quoting is to provide forward references to objects perhaps before they are defined. There's no reason to do this with building typing objects

Here's the mypy documentation on that. You'll see a particular example where they populate a List with a forward reference, where the quoting only apply to the type variable. That's what I'm asking for here:

https://mypy.readthedocs.io/en/latest/kinds_of_types.html#class-name-forward-references

Reply:
Though may not even want this at all:

> String literal types must be defined (or imported) later in the same module. They cannot be used to leave cross-module references unresolved. (For dealing with import cycles, see Import cycles.)

Don't need this to be a blocker

Reply:
Another option here if we don't like the import machinery is to just move this to generic where it is actually used. I don't see a very strong reason why this needs to be in base

Reply:
I've moved ``whitelist_method_generator`` to generic.py. That does make the typing simpler. I've also added the TypeVar in _typing.

However, I suggest that unless we're fixing relatively simple typing issues (no new defined types in ``_typing.py``  etc.), that typing changes go in separate commits?

Reply:
separate commits are certainly fine for typing but not a PR unless major work needs to be done; we get much more mileage out of adding typing as we going in changed code (granted sometimes this will be slightly painful at the beginning)